final_task/rss_reader/rss_parser.py

```diff
@@ -0,0 +1,325 @@
+            print("News were written in html file")
+            logging.debug("News were written in html file")
+        except Exception:
+            logging.error(Exception)
+        finally:
+            f.close()
+        return None
+
+    def printCashe(self):
+        """ Prints news from database from the specified day."""
+
+        logging.debug("Check in printCashe")
+        con = None
+        exists = checkExistance()
+        if not exists:
+            createTable()
+        try:
+            con = psycopg2.connect(
+                database="postgres",
+                user="postgres",
+                password=readingPasswordFromFile(),
+                host="127.0.0.1",
+                port="5432"
+            )
+            logging.debug("Database opened successfully")
+            cur = con.cursor()
+            if self.limit > 0:
+                cur.execute("SELECT * from cache WHERE feed = %s AND date = %s LIMIT %s",
+                            (self.url, self.date, str(self.limit)))
+            else:
+                cur.execute("SELECT * from cache WHERE feed = %s AND date = %s", (self.url, self.date))
+
+            rows = cur.fetchall()
+            if not rows:
+                print("No results\nTry to enter another date or url")
+                return None
+            if self.jsonFormat:
+                self.jsonFromCashe(rows)
+            elif self.to_html:
+                self.htmlFromCashe(rows)
+            else:
+                for row in rows:
+                    print("Title: ", row[1])
+                    print("Date: ", row[2])
+                    print("Link: ", row[4][0])
+                    print()
+                    print("Description: " + row[3])
+                    print("\nLinks:")
+                    print("[1]:", row[4][0])
+                    if len(row[4]) > 1:
+                        print("[2]:", row[4][1])
+                    print("__________________________________________________________________")
+        except (psycopg2.DatabaseError) as error:
+            logging.error(error)
+        except Exception as error:
+            logging.error(error)
+        finally:
+            if con is not None:
+                con.close()
+        return None
+
+    def convertIntoHtmlFormat(self, thefeed):
+        """ Creates html file in specified directory and puts information in database"""
+
+        # openning html-file to write
+        filename = self.path + r"\news.html"
+        try:
+            f = open(filename, 'w')
+        except OSError:
+            logging.error("File can't be opened")
+            print("Could not open file ", filename)
+            return None
+        logging.debug("File was opened successfully")
+
+        doc, tag, text, line = Doc().ttl()
+        line('h1', 'News from ' + thefeed.feed.get("title", ""))
+
+        for index, thefeedentry in enumerate(thefeed.entries):
+            if (index < self.limit) | (self.limit == -1):
```

Comment:
It is better to use `slices` instead of comparing index to limit each on iteration

Reply:
Fixed